Px.Utils/Validation/DataValidation/DataValidatorFunctions.cs

```diff
@@ -0,0 +1,200 @@
+﻿using PxUtils.Validation;
+
+namespace PxUtils.Validation.DataValidation;
+
+/// <summary>
+/// Class for validating the count of rows in a data set.
+/// </summary>
+public class DataRowCountValidator(int numOfRows): IDataValidator
+{
+    private int _currentRow;
+
+    /// <summary>
+    /// Validates the token based on the number of rows specified.
+    /// </summary>
+    /// <param name="token">The token to validate.</param>
+    /// <returns>A collection of validation feedbacks.</returns>
+    public IEnumerable<ValidationFeedback> Validate(Token token)
+    {
+        if (token.Type == TokenType.LineSeparator)
+            _currentRow++;
+        else if (token.Type == TokenType.EndOfStream && _currentRow != numOfRows)
+        {
+            return new[]
+            {
+                new ValidationFeedback(ValidationFeedbackLevel.Error,
+                    ValidationFeedbackRule.DataValidationFeedbackInvalidRowCount, token.LineNumber,
+                    token.CharPosition, $"{numOfRows},{_currentRow}")
+            };
+        }
+
+        return Array.Empty<ValidationFeedback>();
+    }
+}
+
+public class DataRowLengthValidator(int rowLen) : IDataValidator
+{
+    private int _currentRowLen = 0;
+
+    /// <summary>
+    /// Validates the given token for row length.
+    /// </summary>
+    /// <param name="token">The token to validate.</param>
+    /// <returns>An enumerable collection of ValidationFeedback objects.</returns>
+    public IEnumerable<ValidationFeedback> Validate(Token token)
+    {
+        
+        switch (token.Type)
+        {
+            case TokenType.DataItemSeparator:
+                _currentRowLen++;
+                break;
+            case TokenType.LineSeparator:
+                var itemNum = _currentRowLen;
+                if (_currentRowLen != rowLen)
+                {
+                    _currentRowLen = 0;
+                    return new[] { new ValidationFeedback(ValidationFeedbackLevel.Error, 
+                        ValidationFeedbackRule.DataValidationFeedbackInvalidRowLength, token.LineNumber, token.CharPosition,$"{rowLen},{itemNum}") };
+                }
+                _currentRowLen = 0;
+                break;
+        }
+
+        return Array.Empty<ValidationFeedback>();
+    }
+}
+
+public class DataStringValidator : IDataValidator
+{
+    private static readonly string[] ValidStringDataItems = new[]
+        { "\".\"", "\"..\"", "\"...\"", "\"....\"", "\".....\"", "\"......\"", "\"-\"" };
```

Comment:
I'd suggest moving these hard coded syntax strings to PxFileSyntaxConf just in case.

Reply:
will do